### PR TITLE
(maint) Add support for static Boost with ICU

### PR DESCRIPTION
**Pull request by MikaelSmith**
Pulls in the FindICU module from
https://github.com/julp/FindICU.cmake#6fc73154d70b6611027f5475296386c3e66851af

Enabled by setting LEATHERMAN_USE_ICU.

### CODE REVIEW

File: locale/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,19 @@
 if (LEATHERMAN_USE_LOCALES)
     find_package(Boost 1.54 REQUIRED COMPONENTS locale system)
+    if (BOOST_STATIC AND LEATHERMAN_USE_ICU)
+        find_package(ICU)
```
Comment (beav) [inline locale/CMakeLists.txt]: I had to alter this to the following to get it to work for me:

`find_package(ICU COMPONENTS i18n uc)`

Without this, the compile kept forgetting to add `-licu18n`, causing linker errors.

Comment (MikaelSmith) [inline locale/CMakeLists.txt]: Ah, thanks for the feedback! I hadn't actually tried using it, always the best way to add code.

**Comment by McdonaldSeanp** [comment]
jenkins please test facter

**Comment by MikaelSmith** [comment]
That's pretty cool.

**Comment by McdonaldSeanp** [comment]
Yessir it is.

**Comment by McdonaldSeanp** [comment]
jenkins please test facter

**Comment by McdonaldSeanp** [comment]
That last build failure looks legitimate: 

```
14:09:15 1: ...............................................................................
14:09:15 1: 
14:09:15 1: /var/tmp/tmp.Czsw7aqUWV/leatherman/ruby/tests/api-test.cc:185: FAILED:
14:09:15 1: due to unexpected exception with message:
14:09:15 1:   symbol rb_cFixnum was not found in /opt/puppetlabs/puppet/lib/libruby.so.2.4
14:09:15 1:   .2.
14:09:15 1: 
14:09:15 1: ===============================================================================
14:09:15 1: test cases:   86 |   77 passed | 9 failed
14:09:15 1: assertions: 1349 | 1340 passed | 9 failed
14:09:15 1: 
14:09:15 1/1 Test #1: leatherman tests .................***Failed    2.17 sec
14:09:15 
14:09:15 0% tests passed, 1 tests failed out of 1
14:09:15 
14:09:15 Total Test time (real) =   2.17 sec
14:09:15 
14:09:15 The following tests FAILED:
14:09:15 	  1 - leatherman tests (Failed)
```

**Comment by MikaelSmith** [comment]
Ah, this PR's probably really old. I can rebase it.

**Comment by MikaelSmith** [comment]
jenkins please test facter

**Pull request merged**